packages/smol/components/sections/Revoke/useAllowances.tsx

```diff
@@ -0,0 +1,584 @@
+		set_cachedApproveEvents(entriesFromDB);
+	}, [entryNonce, getAll]);
+
+	/**********************************************************************************************
+	 ** Every time we interact with chain-sync events in DB, we want to immedeatley have up to date
+	 ** data from DB
+	 *********************************************************************************************/
+	useAsyncTrigger(async () => {
+		chainSyncNonce;
+		const entryFromDB = await getAllChainSync();
+		set_cachedChainSync(entryFromDB);
+	}, [chainSyncNonce, getAllChainSync]);
+
+	/**********************************************************************************************
+	 ** We're retrieving an array of addresses from the currentNetworkTokenList, intending to
+	 ** obtain allowances for each address in the list. This process allows us to gather allowance
+	 ** data for all tokens listed.
+	 *********************************************************************************************/
+	const tokenAddresses = useMemo(() => {
+		return isTokensLoading ? [] : listTokensWithBalance(currentChainSyncEntry?.chainID).map(item => item.address);
+	}, [currentChainSyncEntry?.chainID, isTokensLoading, listTokensWithBalance]);
+
+	/**********************************************************************************************
+	 ** The allowances vary across different chains, necessitating us to reset the current state
+	 ** when the user switches chains or change the address.
+	 *********************************************************************************************/
+	useAsyncTrigger(async () => {
+		if (
+			!cachedChainSync ||
+			cachedChainSync.some(item => item.address === address && item.chainID === safeChainID) ||
+			!address
+		) {
+			return;
+		}
+
+		addChainSyncEntry({address, chainID: isDev ? chainID : safeChainID, blockNumber: 0n});
+		set_entryNonce(nonce => nonce + 1);
+		set_chainSyncNonce(nonce => nonce + 1);
+	}, [cachedChainSync, address, addChainSyncEntry, chainID, safeChainID]);
+
+	/**********************************************************************************************
+	 ** In DB we store approve-events for all addresses and ChainIDs. But we need to show to user
+	 ** only those that match their address and chainID
+	 *********************************************************************************************/
+	const chainFilteredAllowances = useMemo(() => {
+		const _formatedAllowances: TExpandedAllowance[] = [];
+		for (const allowance of cachedApproveEvents) {
+			_formatedAllowances.push({
+				...allowance,
+				args: {
+					owner: allowance.owner,
+					sender: allowance.sender,
+					value: allowance.value
+				}
+			});
+		}
+
+		const filteredAllowances = _formatedAllowances.filter(
+			item => item.args.owner === address && item.chainID === safeChainID
+		);
+
+		return filteredAllowances;
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [address, `${cachedApproveEvents}`, safeChainID]);
```

Comment:
This looks weird despite it might be working 😄
![image](https://github.com/SmolDapp/SmolV2/assets/32621609/322be24a-8051-4017-ab9c-630322643fa1)


i see few options:
1. use cachedApproveEvents.length (not very proficient too)
2. implement new toJSON method (might be useful actually) - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#use_within_json